lib/tasks/conditions/process_conditions.rake

```diff
@@ -32,7 +32,7 @@ namespace :shf do
 
         if use_slack_notification
           SHFNotifySlack.notify_after(class_name) do
-            process_klass(klass, condition, log)
+            process_klass(klass, condition, log, use_slack_notification: use_slack_notification)
```

Comment:
I could be wrong but it appears there might be a logic problem in the slack-notification error handling.

The rescue clause sets `use_slack_notification = false`, logs error(s) and then calls `retry`.

Subsequently line 38 is called:
```
process_klass(klass, condition, log)
```

The signature of that method:
```
def process_klass(klass, condition, log, use_slack_notification: true)
```
will result in calling the specific condition's `condition_response` method with `use_slack_notification` set too true.

Reply:
Good catch.